validator/client/src/main/java/tech/pegasys/teku/validator/client/KeyManager.java

```diff
@@ -92,8 +100,41 @@ public DeleteKeysResponse deleteValidators(final List<BLSPublicKey> validators)
    * @param slashingProtection a combined slashing protection payload
    * @return a list of 1 status per keystore that was attempted to be imported
    */
-  public List<ImportStatus> importValidators(
+  public List<PostKeyResult> importValidators(
       final List<String> keystores, final List<String> passwords, final String slashingProtection) {
-    throw new NotImplementedException("importValidators not implemented yet");
+
+    List<PostKeyResult> postKeyResults = new ArrayList<>();
+
+    if (keystores.size() == passwords.size()) {
+      final Iterator<String> keystoreIterator = keystores.iterator();
+      final Iterator<String> passwordIterator = passwords.iterator();
+      while (keystoreIterator.hasNext() && passwordIterator.hasNext()) {
+        try {
+          final String password = passwordIterator.next();
+          final KeyStoreData keystore = getKeystoreDataObject(keystoreIterator.next());
+          if (KeyStore.validatePassword(password, keystore)) {
```

Comment:
Instead of using the `validatePassword` we can use `decrypt`, and use it to load the validators in memory. Trying to make the operation more lightweight.